sqla_vertica_python/vertica_python.py

```diff
@@ -362,8 +362,25 @@ def get_check_constraints(self, connection, table_name, schema=None, **kw):
     # constraints are enforced on selects, but returning nothing for these
     # methods allows table introspection to work
 
-    def get_pk_constraint(self, bind, table_name, schema, **kw):
-        return {'constrained_columns': [], 'name': 'undefined'}
+    @reflection.cache
+    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        query = None
+        query = None
```

Comment:
you can delete both of these `query` variables.

Also, I know that you are copying the other query generation from earlier in the file, but it would be much cleaner if this was just a single variable declaration and the schema clause was optionally append. You can one-line this if you'd like

Reply:
Done, and added one variable declaration. Any other suggestions welcome.